integration_test/agents/agents.go

```diff
@@ -874,6 +935,43 @@ func CommonSetupWithExtraCreateArgumentsAndMetadata(t *testing.T, imageSpec stri
 	return ctx, logger, vm
 }
 
+// InstallOpsAgentUAPPluginFromGCS installs the Ops Agent plugin tarball from GCS onto the given Linux VM.
+//
+// gcsPath must point to a GCS Path to a .tar.gz file to install on the testing VMs.
+func InstallOpsAgentUAPPluginFromGCS(ctx context.Context, logger *log.Logger, vm *gce.VM, gcsPath string) error {
+	if gce.IsWindows(vm.ImageSpec) {
+		return fmt.Errorf("Ops Agent UAP Plugin does not support Windows yet")
+	}
+	if _, err := gce.RunRemotely(ctx, logger, vm, "mkdir -p /tmp/agentUpload /tmp/agentPlugin"); err != nil {
+		return err
+	}
+	if err := gce.InstallGsutilIfNeeded(ctx, logger, vm); err != nil {
+		return err
+	}
+	if err := gce.InstallGrpcurlIfNeeded(ctx, logger, vm); err != nil {
+		return err
+	}
+
+	if _, err := gce.RunRemotely(ctx, logger, vm, "sudo gsutil cp -r "+gcsPath+"/* /tmp/agentUpload"); err != nil {
+		return fmt.Errorf("error copying down agent package from GCS: %v", err)
+	}
+	if _, err := gce.RunRemotely(ctx, logger, vm, "mv /tmp/agentUpload/*.tar.gz /tmp/agentPlugin || echo nothing to move"); err != nil {
```

Comment:
We need to error here if there is no UAP plugin .tar.gz at this point. The || echo is preventing the command for erroring out.